### PR TITLE
Xquartz beta

### PR DESCRIPTION
**Pull request by N4M3Z**
Added XQuartz beta and adjusted appcast stream accordingly

### CODE REVIEW

**Comment by radeksimko** [comment]
Hi @N4M3Z , could you please squash your 3 commits into one & rebase latest commits from upstream and push again to see if it's passing all tests?

Thank you for your contribution.

**Comment by N4M3Z** [comment]
Done, hopefully it will pass now.

**Comment by radeksimko** [comment]
The rebasing is all right, but the download server seems to be using a strange certificate:

```
curl -v -I -L https://xquartz.macosforge.org/downloads/SL/XQuartz-2.7.8_beta3.dmg
```
```
* Hostname was NOT found in DNS cache
*   Trying 17.251.224.138...
* Connected to xquartz.macosforge.org (17.251.224.138) port 443 (#0)
* SSL certificate problem: Invalid certificate chain
* Closing connection 0
curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

Does it work for you?

**Comment by N4M3Z** [comment]
Works for me just fine:
```
* Hostname was NOT found in DNS cache
*   Trying 17.251.224.138...
* Connected to xquartz.macosforge.org (17.251.224.138) port 443 (#0)
* TLS 1.2 connection using TLS_RSA_WITH_RC4_128_SHA
* Server certificate: www.macosforge.org
* Server certificate: Symantec Class 3 EV SSL CA - G3
* Server certificate: VeriSign Class 3 Public Primary Certification Authority - G5
> HEAD /downloads/SL/XQuartz-2.7.8_beta3.dmg HTTP/1.1
> User-Agent: curl/7.37.1
> Host: xquartz.macosforge.org
> Accept: */*
```

Although if I do just:
```
curl https://xquartz.macosforge.org/downloads/SL/XQuartz-2.7.8_beta3.dmg
```
then it prints a notification that the file moved: https://xquartz-dl.macosforge.org/SL/XQuartz-2.7.8_beta3.dmg. However, I think we should use the original link provided for the cask.

**Comment by radeksimko** [comment]
It does work for me now (it didn't yesterday).
I'll just go through all the stanzas there to make sure they're all correct, bear with me.

File: Casks/xquartz-beta.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Pathname.new(File.expand_path('~')).join('Library', 'Logs').mkpath
+
+    # Set default path to X11 to avoid the need of manual setup
+    system '/usr/bin/defaults', 'write', 'com.apple.applescript', 'ApplicationMap', '-dict-add', 'X11', 'file://localhost/Applications/Utilities/XQuartz.app/'
+
+    # Load & start XServer to avoid the need of relogin
+    system '/bin/launchctl', 'load', '/Library/LaunchAgents/org.macosforge.xquartz.startx.plist'
+
+    # Set automatic Updates for Beta Versions
+    system 'defaults', 'write', 'org.macosforge.xquartz.X11', 'SUFeedURL', 'http://xquartz.macosforge.org/downloads/sparkle/beta.xml'
```
Comment (radeksimko) [inline Casks/xquartz-beta.rb]: The update URL is already contained in the `Info.plist`, are you sure this is necessary?

```
$ /usr/libexec/PlistBuddy -c 'Print :SUFeedURL' Info.plist
http://xquartz.macosforge.org/downloads/sparkle/beta.xml
```

File: Casks/xquartz-beta.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  uninstall :quit => 'org.macosforge.xquartz.X11',
+            :launchctl => 'org.macosforge.xquartz.startx',
```
Comment (radeksimko) [inline Casks/xquartz-beta.rb]: It seems it is creating two files/records for launchctl:
```
$ ./developer/bin/list_installed_launchjob_ids
```
```
org.macosforge.xquartz.privileged_startx
org.macosforge.xquartz.startx
```

Could you change this line to `:launchctl => 'org.macosforge.xquartz.*',` ?

**Comment by radeksimko** [comment]
Besides the two comments, it LGTM.

**Comment by fanquake** [comment]
@N4M3Z Do you plan on updating this?

**Comment by N4M3Z** [comment]
Yes

File: Casks/xquartz-beta.rb (hunk 1 above)
Comment (N4M3Z) [inline Casks/xquartz-beta.rb]: Not sure, I just followed the implementation of the xquartz cask

**Comment by radeksimko** [comment]
Great job, now could you please just squash these two commits into one?

In case you already know how to do it, just skip everything below this line :wink: 

------

Being on your branch:
```
git rebase HEAD~2 -i
```
then you gonna see something like this:
```
pick b91c04b Added formula for XQuartz beta release
pick b95562f Updated uninstall as per comment from radeksimko
```
so change the action for the second commit to squash:
```
pick b91c04b Added formula for XQuartz beta release
squash b95562f Updated uninstall as per comment from radeksimko
```
and save + quit.

Then add a correct commit message, e.g. _`XQuartz 2.7.8_beta3 added (new cask)`_
and finally `git push origin xquartz-beta --force` (`--force` because you're effectively rewriting history on your branch).

**Comment by N4M3Z** [comment]
I got familiar with squashing when addressing your original comment. Anyway, is there any difference to using --force/+xquartz-beta when pushing the rewritten history to repo?

**Comment by visualphoenix** [comment]
:+1:

**Comment by fanquake** [comment]
Wanted to get this merged. Improvements can made be in future.

**Pull request merged**